ui/src/tempVars/components/KeysTemplateBuilder.tsx

```diff
@@ -0,0 +1,241 @@
+  measurements: string[]
+  measurementsStatus: RemoteDataState
+  selectedMeasurement: string
+  keys: string[]
+  keysStatus: RemoteDataState
+}
+
+@ErrorHandling
+export default class KeysTemplateBuilder extends PureComponent<Props, State> {
+  constructor(props) {
+    super(props)
+
+    const selectedDatabase = _.get(props, 'template.query.db', '')
+    const selectedMeasurement = _.get(props, 'template.query.measurement', '')
+
+    this.state = {
+      databases: [],
+      databasesStatus: RemoteDataState.Loading,
+      selectedDatabase,
+      measurements: [],
+      measurementsStatus: RemoteDataState.Loading,
+      selectedMeasurement,
+      keys: [],
+      keysStatus: RemoteDataState.Loading,
+    }
+  }
+
+  public async componentDidMount() {
+    await this.loadDatabases()
+    await this.loadMeasurements()
+    await this.loadKeys()
+  }
+
+  public render() {
+    const {queryPrefix} = this.props
+    const {
+      databases,
+      databasesStatus,
+      selectedDatabase,
+      measurements,
+      measurementsStatus,
+      selectedMeasurement,
+      keys,
+      keysStatus,
+    } = this.state
+
+    return (
+      <div className="temp-builder measurements-temp-builder">
+        <div className="form-group">
+          <label>Meta Query</label>
+          <div className="temp-builder--mq-controls">
+            <div className="temp-builder--mq-text">{queryPrefix}</div>
+            <DropdownPlaceholder rds={databasesStatus}>
+              <Dropdown
+                items={databases.map(text => ({text}))}
+                onChoose={this.handleChooseDatabaseDropdown}
+                selected={selectedDatabase}
+                buttonSize=""
+              />
+            </DropdownPlaceholder>
+            <div className="temp-builder--mq-text">FROM</div>
+            <DropdownPlaceholder rds={measurementsStatus}>
+              <Dropdown
+                items={measurements.map(text => ({text}))}
+                onChoose={this.handleChooseMeasurementDropdown}
+                selected={selectedMeasurement}
+                buttonSize=""
+              />
+            </DropdownPlaceholder>
+          </div>
+        </div>
+        <TemplateMetaQueryPreview items={keys} loadingState={keysStatus} />
+      </div>
+    )
+  }
+
+  private async loadDatabases() {
+    const {source} = this.props
+
+    this.setState({databasesStatus: RemoteDataState.Loading})
+
+    try {
+      const {data} = await showDatabases(source.links.proxy)
+      const {databases} = parseShowDatabases(data)
+      const {selectedDatabase} = this.state
+
+      this.setState({
+        databases,
+        databasesStatus: RemoteDataState.Done,
+      })
+
+      if (!selectedDatabase) {
+        this.handleChooseDatabase(_.get(databases, 0, ''))
+      }
+    } catch (error) {
+      this.setState({databasesStatus: RemoteDataState.Error})
+      console.error(error)
+    }
+  }
+
+  private async loadMeasurements() {
```

Comment:
could you type-annotate this? even with `void` if no return. sooner than later we'll flip on `no-implicit-any`